lib/CL/devices/builtin_kernels.cc

```diff
@@ -214,6 +214,44 @@ BIKD pocl_BIDescriptors[BIKERNELS] = {
              BIArg("unsigned int*", "minloc", WRITE_BUF),
              BIArg("unsigned int*", "maxloc", WRITE_BUF),
          }),
+    BIKD(POCL_CDBI_SOBEL3X3_U8,
```

Comment:
We should move this whole file to a C99 implementation. Now this forces us to build "nano-PoCL" with C++ if we want to include built-in kernels. If it's easy to do so, please do with this commit. Otherwise let's put this high in the laundry list.

Reply:
postponing this since this will need a refactor anyway related to the DBK-work